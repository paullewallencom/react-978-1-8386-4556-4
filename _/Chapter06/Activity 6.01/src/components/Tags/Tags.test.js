import React, { useState } from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Tags } from "./index";

afterEach(cleanup);

const tags = ["black", "chocolate", "milk", "froth", "water"];

it("renders a list of tags", () => {
  const { asFragment } = render(
    <Tags tags={tags} selectedTag={null} handleTags={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});

const TagDemo = () => {
  const [selectedTag, setSelectedTag] = useState(null);
  return (
    <Tags
      tags={tags}
      selectedTag={selectedTag}
      handleTags={tag => () => {
        setSelectedTag(tag);
      }}
    />
  );
};

it("sets active tag when a tag is clicked", () => {
  const { container, getByText } = render(<TagDemo />);
  expect(container.querySelectorAll(".active").length).toBe(0);
  const selectedTag = "milk";
  fireEvent.click(getByText(selectedTag));
  expect(container.querySelectorAll(".active").length).toBe(1);
  expect(getByText(selectedTag).classList.contains("active")).toBe(true);
});
