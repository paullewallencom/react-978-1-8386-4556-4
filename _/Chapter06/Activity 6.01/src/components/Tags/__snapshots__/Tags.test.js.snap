// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of tags 1`] = `
<DocumentFragment>
  <div>
    Select Filter
    <button
      class="tag "
    >
      black
    </button>
    <button
      class="tag "
    >
      chocolate
    </button>
    <button
      class="tag "
    >
      milk
    </button>
    <button
      class="tag "
    >
      froth
    </button>
    <button
      class="tag "
    >
      water
    </button>
  </div>
</DocumentFragment>
`;
